### PR TITLE
Expose current log file info in `didLog` methods.

### PR DESCRIPTION
**Pull request by hhanesand**
I removed a public variable to provide queue protection in d0fed54ea39d710918f5b9d3627d32a50f889c46 .

### CODE REVIEW

File: Classes/DDFileLogger.m [hunk 1]
```diff
@@ -1217,13 +1225,23 @@ - (void)lt_logData:(NSData *)data {
     }
 
     @try {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         [self willLogMessage];
```
Comment (ffried) [inline Classes/DDFileLogger.m]: We should only call this, if the new version is not implemented.

File: Classes/DDFileLogger.m [hunk 2]
```diff
@@ -1217,13 +1225,23 @@ - (void)lt_logData:(NSData *)data {
...
+
+        [self willLogMessage:_currentLogFileInfo];
 
         NSFileHandle *handle = [self lt_currentLogFileHandle];
         [handle seekToEndOfFile];
         [handle writeData:data];
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         [self didLogMessage];
```
Comment (ffried) [inline Classes/DDFileLogger.m]: same here

**Comment by OSSPolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any changes to library code should be reflected in the CHANGELOG. Please consider adding a note there about your change.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The library files were changed, but the tests remained unmodified. Consider updating or adding to the tests to match the library changes.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by hhanesand** [comment]
This would crash if subclasses called the deprecated methods since we were no longer implementing it. Fixed by forwarding to `dummyMethod`. I've checked that `respondsToSelector:` still returns `NO` even though we are forwarding the methods.

**Comment by hhanesand** [comment]
@ffried

**Comment by ffried** [comment]
@hhanesand Hmm, I'm not sure if this isn't causing more confusion than it should. When I wrote my comment, I didn't consider the fact, that we need to implement it ourselves, too.

Couldn't we just implement the two methods (empty) and add `[self class] != [DDFileLogger class]` to the `respondsToSelector:` checks?
So that `implementsDeprecated{Will,Did}Log` is only true, if we're being subclassed.

**Comment by hhanesand** [comment]
I don't think that would work?

If we are subclassed, and we implement the deprecated methods, then `respondsToSelector:` will always return `YES`. We'd need a way to test if the subclass overrides the method. Changing the check to  `respondsToSelector: && [self class] != [DDFileLogger]` would mean we always call the deprecated method, regardless of the subclass's overriding state. Maybe I'm not understanding.

This way, we don't implement the deprecated methods, so `repondsToSelector:` performs the "does the subclass implement method" check. Furthermore, to avoid crashes, we forward any `[super will/didLogMessage]` calls that may or may not exist into another method.

**Comment by hhanesand** [comment]
@ffried

**Comment by ffried** [comment]
@hhanesand You're absolutely right. Not my best day today, sorry!
There are ways to get the methods of a subclass, but that would definitively be more complicated than what you've done now.

Just to jumpstart my slow brain today: Why do we need to override `forwardInvocation:`?

**Comment by hhanesand** [comment]
`[NSObject forwardInvocation:]` docs :
```
To respond to methods that your object does not itself recognize, you must override methodSignatureForSelector: in addition to forwardInvocation:. The mechanism for forwarding messages uses information obtained from methodSignatureForSelector: to create the NSInvocation object to be forwarded. Your overriding method must provide an appropriate method signature for the given selector, either by pre formulating one or by asking another object for one.
```

You can see it being used in ``DDFileLogger+Buffering.m` as well. We may not actually need to override `forwardInvocation:`, haven't tested.

**Comment by ffried** [comment]
@hhanesand Ah right, I see. Should we check the `invocation.selector` and `invocation.target` if it's our dummy method and otherwise invoke `[super forwardInvocation:invocation]`?
At least in the sample in the docs, they do something similar.

**Comment by hhanesand** [comment]
Yeah, that'd probably be wise.

**Comment by ffried** [comment]
Great. Thanks for your patience today with me 😀
LGTM now. 👍

**Comment by hhanesand** [comment]
No problem! Happens to all of us :)

**Pull request merged**